Carrying system.

Adrian16199 opened the pull request:
Carrying someone in your arms, maybe your beloved or a high criminal, is nice for both.

IT WORKS (Thanks death, you are cool x3)

<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
So far, only added, might need to change 3 files with 1 liners

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
Because carrying someone is very important for ro- I mean to help them traverse.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Uh, i stole it x3

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
- add: Carrying someone in your arms.
- remove: Not being able to do that.

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Clothing/Masks/masks.yml:
```diff
@@ -0,0 +1,21 @@
+- type: entity
+  parent: ClothingMaskPullableBase
+  id: ClothingMaskHockeyBase
+  description: It's a dirty looking hockey mask, with some red markings on it, presumably painted. Whoever wore this was an aggressive player.
```
```suggestion
  description: It's a dirty-looking hockey mask, with some red markings on it, presumably painted. Whoever wore this was an aggressive player.
```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Clothing/Uniforms/jumpsuits.yml:
```diff
@@ -9,6 +9,17 @@
+- type: entity
+  parent: ClothingUniformBase
+  id: ClothingUniformJumpsuitBloodyClothes
+  name: blood soaked clothes
```
```suggestion
  name: blood-soaked clothes
```

DEATHB4DEFEAT reviewed (changes requested):
grammar police

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Clothing/Masks/masks.yml:
```diff
@@ -0,0 +1,21 @@
+- type: entity
+  parent: ClothingMaskPullableBase
+  id: ClothingMaskHockeyBase
+  description: It's a dirty-looking hockey mask, with some red markings on it, presumably painted. Whoever wore this was an aggressive player.
+  abstract: true
+  components:
+  - type: IngestionBlocker
+  - type: DiseaseProtection
+    protection: 0.05
+  - type: IdentityBlocker
+  - type: BreathMask
```
Why use an abstract type? There shouldn't be a description on it either.

ps3moira replied:
I was just following what Nyano-Code masks have, since it seems to not work if you don't have that part included.

Adrian16199 committed:
Carrying system.

Carrying someone in your arms, maybe your beloved or a high criminal, is nice for both.

TO DO-Pretty much make it work

Adrian16199 committed:
Many more additions to actualy make it work...

Halfly

Adrian16199 committed:
Merge branch 'master' into Carrying_system

Signed-off-by: Adrian16199 <144424013+Adrian16199@users.noreply.github.com>

Adrian16199 committed:
ITS WORKING!!!

Not done yet, need to improve it

Adrian16199 committed:
THE CARRIABLE UPDATE!!!!!

YEEEEEEEEEEEEEEEEEEEEEEEEEEE
(Thanks death for helping with that one issue that I wasnt able to figure out x3)

Adrian16199 committed:
Merge branch 'master' into Carrying_system

Adrian16199 committed:
Merge branch 'master' into Carrying_system

luringens reviewed (commented):
I see Vulpkanin got the `- type: Carriable` component, but not the others? At least I didn't spot it in these three:
- Resources/Prototypes/DeltaV/Entities/Mobs/Species/harpy.yml
- Resources/Prototypes/Nyanotrasen/Entities/Mobs/Species/felinid.yml
- Resources/Prototypes/Nyanotrasen/Entities/Mobs/Species/Oni.yml
Or maybe it should be in their common Resources/Prototypes/Entities/Mobs/Species/human.yml, where they seem to get their hunger/thirst values from? I'm new to the codebase, so might just be missing something 🙂

Adrian16199 committed:
Merge branch 'master' into Carrying_system

Adrian16199 committed:
Merge branch 'master' into Carrying_system
